src/OpenLoco/EditorController.cpp

```diff
@@ -35,16 +57,232 @@ namespace OpenLoco::EditorController
+                break;
+
+            case Step::scenarioOptions:
+                // 0x0043D12C
+                WindowManager::closeAllFloatingWindows();
+                TerraformConfig();
+                _editorStep = Step::landscapeEditor;
+                LandscapeGeneration::open();
+                break;
+
+            case Step::saveScenario:
+                break;
+        }
+
+        Gfx::invalidateScreen();
+    }
+
+    void ClimateInit()
+    {
+        call(0x00496A18);
     }
 
     // 0x0043D15D
     void goToNextStep()
     {
-        call(0x0043D15D);
+        switch (getCurrentStep())
+        {
+            case Step::null:
+                break;
+
+            case Step::objectSelection:
+                call(0x473A13);
+                call(0x4747D4);
+                call(0x4748D4);
+                ClimateInit();
+                TerraformConfig();
+                _editorStep = Step::landscapeEditor;
+                LandscapeGeneration::open();
+                if (S5::getOptions().scenarioFlags & Scenario::landscape_generation_done)
+                {
+                    if ((addr<0x00525E28, uint32_t>() & 1) == 0)
+                    {
+                        Scenario::generateLandscape();
+                    }
+                }
+                break;
+
+            case Step::landscapeEditor:
+            {
+                if (!validateStep1())
+                {
+                    Windows::Error::open(StringIds::str_1672, gGameCommandErrorText);
+                    break;
+                }
+
+                int eax = S5::getOptions().objectiveDeliveredCargoType;
+                if (ObjectManager::get<cargo_object>(eax) == nullptr)
+                {
+                    for (size_t i = 0; i < ObjectManager::getMaxObjects(object_type::cargo); i++)
+                    {
+                        if (ObjectManager::get<cargo_object>(i) != nullptr)
+                        {
+                            S5::getOptions().objectiveDeliveredCargoType = static_cast<uint8_t>(i);
+                            break;
+                        }
+                    }
+                }
+
+                WindowManager::closeAllFloatingWindows();
+                Scenario::initialiseDate(S5::getOptions().scenarioStartYear);
+                ClimateInit();
+                ScenarioOptions::open();
+                _editorStep = Step::scenarioOptions;
+                break;
+            }
+
+            case Step::scenarioOptions:
+            {
+                sub_440165();
+                // Original checks for a non-true response, and aborts saving with an error.
+
+                WindowManager::closeAllFloatingWindows();
+
+                if (!sub_4418DB())
+                    break;
+
+                _editorStep = Step::null;
+                call(0x0046F910);
+
+                auto path = fs::path(scenarioFilename.get());
+                path.replace_extension(".SC5");
```

Comment:
This might fix #802. Side note:
```suggestion
                path.replace_extension(S5::extensionSC5);
```